csvdiff/__init__.py

```diff
@@ -31,17 +31,26 @@
 EXIT_ERROR = 2
 
 
+def diff(from_csv, to_csv, index_columns, sep=',', ignored_columns=None):
+    """
+    Diff two CSVs, returning the patch which transforms one into the
+    other.
+    """
+    return diff_files(io.StringIO(from_csv), io.StringIO(to_csv),
+                      index_columns, sep, ignored_columns)
+
+
 def diff_files(from_file, to_file, index_columns, sep=',', ignored_columns=None):
     """
     Diff two CSV files, returning the patch which transforms one into the
     other.
     """
     with open(from_file) as from_stream:
-        with open(to_file) as to_stream:
-            from_records = records.load(from_stream, sep=sep)
-            to_records = records.load(to_stream, sep=sep)
-            return patch.create(from_records, to_records, index_columns,
-                                ignore_columns=ignored_columns)
+        from_records = records.load(from_stream, sep=sep)
+    with open(to_file) as to_stream:
+        to_records = records.load(to_stream, sep=sep)
+    return patch.create(from_records, to_records, index_columns,
+                        ignore_columns=ignored_columns)
```

Comment:
`from_records` is still just an iterator here, that's why this fails tests in CI

You could do something like this to load it totally into memory (which we will need to do anyway).

```python
with open(from_file) as from_stream:
    from_records = records.load(from_stream, sep=sep)
    from_index = records.index(from_records, index_columns)
...
return patch.create_indexed(from_index, to_index, index_columns, ...)
```